core/src/ast/ast_literal.rs

```diff
@@ -24,6 +24,22 @@ pub enum DateTimeField {
     Hour,
     Minute,
     Second,
+    Century,
+    Decade,
+    Week,
+    Dow, //day of week
+    Doy, //day of year
+    Epoch,
+    Isodow,
+    Isoyear,
+    Julian,
+    Microseconds,
+    Millenium,
+    Milliseconds,
+    Quarter,
+    Timezone,
+    TimezoneHour,
+    TimezoneMinute,
```

Comment:
These are new enum values which GlueSQL does not support yet.

fyi.
`translate` module has two main roles. Second role would be good to consider in this case.

1. Generating executable AST which GlueSQL `executor` can handle
Not like other many SQL databases, GlueSQL's query planner layer is not mandatory.
Storage makers can choose whether to use query planner or not, or using partial planners can be also the choice.
`translate` module converts `sqlparser-rs`'s AST to `gluesql`'s internal AST which also includes a kind of default execution plan.
So, in the `plan` module, it takes AST and returns the same AST (same type, but internal info can be different)

2. Filtering out superset AST values which `sqlparser-rs` supports but `gluesql` does not.
We use `sqlparser-rs` to parse SQL text so we don't need to implement SQL parser by ourselves.
This saves tons of our time, but there's something to consider. `sqlparser-rs` aims to support multiple SQL syntaxes as many as possible. This means that the coverage `sqlparser-rs` has is wider than what GlueSQL supports.
Therefore, it would be good to remove all features & values which GlueSQL does not support in the `translate` module.
Then, theoretically `executor` layer can have zero `Unreachable-` errors. (currently lots of them has removed but still some of them remains)

---
```
    Century,
    Decade,
    Week,
    Dow, //day of week
    Doy, //day of year
    Epoch,
    Isodow,
    Isoyear,
    Julian,
    Microseconds,
```
We can think of adding these values to our AST as a new feature to implement.
Then, it is not necessary to include these in the code before we actually support them.

---
e.g.
https://github.com/gluesql/gluesql/pull/541

Currently, GlueSQL supports `COUNT(*)` wildcard expression but `COUNT(table.*)` is not supported.
Because of this, though `sqlparser-rs` has `FunctionArgExpr::QualifiedWildcard(_)` but GlueSQL AST `CountArgExpr` does not.
```
#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub enum CountArgExpr {
    Expr(Expr),
    Wildcard,
}
```
When we support `COUNT(table.*)`, then that would be the proper time to add `QualifiedWildcard` to `CountArgExpr`.

Reply:
So translate_datetime_field function needs to return a Result<DataTimeField> so that an error can be produced if non supported sql is produced by the user?

Reply:
Okay.. I think I got it now..  Can you take another look?

Thanks!